azavea/echo-locator
# Feature/kak/non routable markers#83

## Overview

Dynamically update neighborhood point GeoJSON map layer to style markers for routable neighborhoods differently from non-routable ones. On click, only go to detail view/select neighborhood for routable neighborhoods.


### Demo

Transit mode:
![image](https://user-images.githubusercontent.com/960264/55210536-3963f900-51be-11e9-842b-32ae8d361079.png)

All reachable in driving mode:
![image](https://user-images.githubusercontent.com/960264/55210445-d2464480-51bd-11e9-85c5-af5d0aa3b20e.png)


### Notes

The map layer had to be wrapped in a layer group for data updates to work via setting the react component `key`, which [means something else](https://github.com/mhasbie/react-leaflet-vectorgrid/blob/master/src/VectorGrid.js#L95) to the Leaflet layer.


## Testing Instructions

 * Reachable neighborhoods should have white circle markers
 * Unreachable neighborhoods should have grey markers
 * Markers are black on hover
 * Clicking a reachable/white marker should go to the details for that neighborhood
 * Clicking an unreachable/grey marker should do nothing in the UI


Closes #83 
Closes #86
